### ISSUE TITLE
ClosingParenthesisIndentation doesn't leave ) to the left edge

### ISSUE THREAD

**Issue by ptarjan** [rubocop-hq/rubocop#2791]
It correctly works for parameters

```
# good
foo(
  a,
  b,
)
```

but it complains about 

```
# incorrect lint error
foo({
  a: b
},
)
```

which rubocop suggests I do

```
# incorrect autofix
foo({
  a: b
},
   )
```

Similarly if I leave a param on the first line, it won't let me left alignt

```
# incorrect lint error
foo(a,
  b,
)
```

If this is intentional behavior, can I get an option in the same vein as 

```
Style/IndentHash:
  EnforcedStyle: consistent
Style/IndentArray:
  EnforcedStyle: consistent
Style/FirstParameterIndentation:
  EnforcedStyle: consistent
Style/MultilineOperationIndentation:
  EnforcedStyle: indented
```

which leaves these on the left?

**Comment by gokulnath** [comment]
Didn't really comprehend the issue. I tested with the code styles above and found that rubocop does correctly autofix those. 
### Case 1
```
# before
foo({
  a: 'b'
}
)
```


```
# after rubocop autofix
foo(a: 'b')
```

### Case 2
```
# before
foo('a',
  'b'
)
```

```
# After rubocop autofix
foo('a',
    'b')
```

Or did you have some doubts regarding the cops?

**Comment by jfelchner** [comment]
@ptarjan my PR for ClosingParenthesisIndentation linked above will fix your "left edge" issue.  It will not however, fix this:

```ruby
foo({
  a: b
},
)
```

which it will still want to autocorrect to this:

```ruby
foo({
  a: b
},
)
```

The reason for this is simple: The cop uses the indentation of the parameter to determine where the `)` should live.  That is its sole responsibility.  By not having a line break after the opening `(`, you're signifying to the cop that you want the closing parenthesis treated in "alignment" mode rather than "outdent" mode.  The snippet above is extremely non-ideomatic Ruby and would almost certainly get caught my a multitude of other cops.  Generally the more ideomatic Ruby example would be:

```ruby
foo(
  a: b
)
```

with the unneeded braces removed from the last item in an argument list.  In this case the cop does the right thing.  Even if you had:

```ruby
foo({
  a: b
})
```

the cop would do the right thing as well.

It would be about double or triple the code to hit that one (very) edge case.  If you agree, please close the issue.  Thanks! :grin:

**Comment by Drenmi** [comment]
> It would be about double or triple the code to hit that one (very) edge case. If you agree, please close the issue. Thanks! 😁

Thanks for looking into the implementation @jfelchner. Since there haven't been any additional reports, I'm closing this for now. 🙂
